src/base/net/downloadmanager.cpp

```diff
@@ -416,6 +418,17 @@ namespace
             delete m_reply;
     }
 
+    void DownloadHandlerImpl::cancel()
+    {
+        if (m_reply) {
+            m_reply->abort();
```

Comment:
should you call `setError();` at here too? or even `setError(); finish();`?

Reply:
No. abort() emits finished() signal with appropriate error.